include/revng/Model/TupleTree.h

```diff
@@ -25,7 +26,7 @@
 template<typename T>
 concept TupleTreeCompatible = IsKeyedObjectContainer<T>
                                 or HasTupleSize<T>
-                                or IsUpcastablePointer<T>;
+                                or UpcastablePointerLike<T>;
```

Comment:
Why?
Is this a self-contained change that we can isolate in a commit?

Reply:
Yup, we can. Before this change it was too strict (`IsUpcastablePointer` is stricter than `UpcastablePointerLike`), but it never failed nevertheless, because we didn't have any instance that was `UpcastablePointerLike` but not `IsUpcastablePointer`. Now it's needed but I'll move it into a separate commit.

Reply:
Done